push.pl

```diff
@@ -211,6 +212,35 @@ sub prompt_for_edit {
 	my $repoDetails = $repoTx->res->json;
 	$repoDetails->{description} //= '';
 	$repoDetails->{full_description} //= '';
+	$repoDetails->{categories} //= [];
+	my @repoCategories = sort map { $_->{slug} } @{ $repoDetails->{categories} };
+	
+	# read local categories from metadata.json
+	my $repoMetadataBytes = Mojo::File->new($repoName . '/metadata.json')->slurp;
+	my $repoMetadataJson = decode_json $repoMetadataBytes;
+	my @localRepoCategories = sort @{ $repoMetadataJson->{hub}{categories} };
+	
+	# check if the local categories differ in length or items from the remote
+	my $needCat = @localRepoCategories != @repoCategories;
+	if (! $needCat) {
+		foreach my $i (0 .. @localRepoCategories) {
+			last if ! defined $repoCategories[$i]; # length difference already covered, so we can bail
+			if ($localRepoCategories[$i] ne $repoCategories[$i]) {
+				$needCat = 1;
+				last;
+			}
+		}
+	}
+	if ($needCat) {
+		say 'updating ' . $repoName . ' categories';
+		my $catsPatch = $ua->patch($repoUrl . 'categories/' => { %$authorizationHeader, Accept => 'application/json' } => json => [
+				map { {
+					slug => $_,
+					name => 'All those moments will be lost in time, like tears in rain... Time to die.',
```

Comment:
A reminder that the `name` property is not used

Reply:
Yeah, it is required to be non-empty, but is unused as far as I can tell.